### PR TITLE
feat(x/ecocredit): add basket curator to state

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #867

Add curator to basket state.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by ryanchristo** [comment]
Waiting on #869

File: proto/regen/ecocredit/basket/v2/state.proto [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    index: {id: 2, fields: "name", unique: true}
+  };
+
+  // id is the uint64 ID of the basket. It is used internally for reducing
+  // storage space.
+  uint64 id = 1;
+
+  // curator is the address of the basket curator who is able to change certain
+  // basket settings.
+  string curator = 2;
```
Comment (ryanchristo) [inline proto/regen/ecocredit/basket/v2/state.proto:26]: Added curator address here.

**Review by technicallyty — APPROVED** [review]
LGTM

**Comment by aaronc** [comment]
Why v2? No need to bump versions over 1 field. We should keep v1 and build upon it. Just make make since and revision comments like in the SDK

**Comment by aaronc** [comment]
See https://github.com/cosmos/cosmos-sdk/blob/master/proto/cosmos/app/v1alpha1/module.proto

**Comment by ryanchristo** [comment]
> Why v2? No need to bump versions over 1 field. We should keep v1 and build upon it. Just make make since and revision comments like in the SDK

Reverted version bump and added since. I'm not seeing revision used in the sdk, only since. Nor is it in the guidelines for proto updates: https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-044-protobuf-updates-guidelines.md. Is there an example of this being used? I was unable to find anything.

**Comment by aaronc** [comment]
> > Why v2? No need to bump versions over 1 field. We should keep v1 and build upon it. Just make make since and revision comments like in the SDK
> 
> Reverted version bump and added since. I'm not seeing revision used in the sdk, only since. Nor is it in the guidelines for proto updates: https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-044-protobuf-updates-guidelines.md. Is there an example of this being used? I was unable to find anything.

No, we should add that to the ADR and it's intended for future use.

**Comment by ryanchristo** [comment]
So would this be the correct format?

```
// Revision 1
package regen.ecocredit.basket.v1;
```

**Comment by ryanchristo** [comment]
And then the message field...

```
    // curator is the address of the basket curator who is able to change certain
    // basket settings.
    //
    // Since: regen-ledger 4.0
    // Since Revision 1
    string curator = 8;
```

**Comment by aaronc** [comment]
Yes and yes 👍

**Comment by ryanchristo** [comment]
Changes added. Can I get a approval/review?

**Review by clevinson — APPROVED** [review]

**Pull request merged**